lib/tasks/jenkins.rake

```diff
@@ -0,0 +1,18 @@
+require 'English'
+
+begin
+  require "ci/reporter/rake/minitest"
+  namespace :jenkins do
+    task :unit => ["jenkins:setup:minitest", 'rake:test']
+
+    namespace :setup do
+      task :pre_ci do
+        ENV["CI_REPORTS"] = 'jenkins/reports/unit/'
+        gem 'ci_reporter'
+      end
+      task :minitest  => [:pre_ci, "ci:setup:minitest"]
+    end
+  end
+rescue LoadError
+  # ci/reporter/rake/rspec not present, skipping this definition
+end
```

Comment:
Perhaps this comes from Python, but I've grown fond of limiting the part that can raise an exception and then putting the real code in `else`.
```suggestion
begin
  require "ci/reporter/rake/minitest"
rescue LoadError
  # ci/reporter/rake/rspec not present, skipping this definition
else
  namespace :jenkins do
    task :unit => ["jenkins:setup:minitest", 'rake:test']

    namespace :setup do
      task :pre_ci do
        ENV["CI_REPORTS"] = 'jenkins/reports/unit/'
        gem 'ci_reporter'
      end
      task :minitest  => [:pre_ci, "ci:setup:minitest"]
    end
  end
end
```

Reply:
I've not seen the pattern before using begin/rescue/else. Structurally though I think it confuses me a bit. As I am not clear reading it if the `else` is tied to the begin or the rescue? And I thought the `namespace :jenkins` block should only happen if the require actually passes?

BTW I borrowed this task from -- https://github.com/theforeman/smart-proxy/blob/develop/tasks/jenkins.rake

Reply:
> I've not seen the pattern before using begin/rescue/else. Structurally though I think it confuses me a bit. As I am not clear reading it if the else is tied to the begin or the rescue? And I thought the namespace :jenkins block should only happen if the require actually passes?

I couldn't find a more recent version of the doc, but https://docs.ruby-lang.org/en/2.4.0/syntax/exceptions_rdoc.html has:
```ruby
begin
  # ...
rescue
  # ...
else
  # this runs only when no exception was raised
ensure
  # ...
end
```

Or from https://stackify.com/rescue-exceptions-ruby/ an even clearer example:

```ruby
begin
    #... process, may raise an exception
rescue =>
    #... error handler
else
    #... executes when no error
ensure
    #... always executed
end
```

It is the block that is executed if no exception was raised.

> BTW I borrowed this task from -- https://github.com/theforeman/smart-proxy/blob/develop/tasks/jenkins.rake

It was a syntax that Ruby introduced somewhere after it was added to Smart Proxy. In my memory it was Ruby 2.5, but that it was documented in 2.4 tells me I'm wrong.

The idea is that any code in could raise an exception, and you want it to be as specific as possible.